File: cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

// Initialize Functions
int greedy(int owed);

int main(void)
{
    // Prompt user for owed cash
    int owed;
    do
    {
        owed = round(get_float("Change owed: ") * 100);
    }
    while (owed < 0);

    // Print answer using greedy algorithm
    printf("%i\n", greedy(owed));
    return 0;
}

// Greedy algorithm
int greedy(int owed)
{
    int answer = 0;
    answer += owed / 25;
    answer += (owed % 25) / 10;
    answer += (owed % 25 % 10) / 5;
    answer += (owed % 5);
    return answer;
}

File: cash.py
from cs50 import get_float
from sys import exit

# Global Variables
quarter = 25
dime = 10
nickel = 5
penny = 1


# Main Function
def main():
    coins = 0
    # Gets a non-negative float from a user
    while True:
        owed = get_float("Change owed: ")
        if owed >= 0:
            break
    # Multiplies and round the owed float to do math with coins
    owed *= 100
    owed = round(owed)

    # Greedy Algorithm
    while owed > 0:
        if owed >= quarter:
            coins += 1
            owed -= quarter
        elif owed >= dime:
            coins += 1
            owed -= dime
        elif owed >= nickel:
            coins += 1
            owed -= nickel
        else:
            coins += 1
            owed -= penny
    # Prints how many coins needed from Greedy Algoritm
    print(coins)
    # Program ran successfully
    exit(0)


main()